### ISSUE TITLE
Sync schemas for ViewSetupHints between API and XML

### ISSUE THREAD

**Issue by GeorgDangl** [buildingSMART/BCF-XML#174]
Related issue:
https://github.com/buildingSMART/BCF-API/issues/164

The data is the same in both the API and XML, but the format is different. In the API, `view_setup_hints` is inside `visibility`, while in XML it's on the same level as `Visibility`. This should be corrected in the XML format.

**Comment by GeorgDangl** [comment]
Suggestion:  For version 3.0, we suggest to have `selection` outside and move `coloring` inside the `visibility` object.

The current situation results from an error, both XML and API contain the same information but have a different structure. This should be prevented in future versions.

**Comment by Wilt** [comment]
I see now that #164 is still up for Release 3.0, which is great!

But I wonder why you want to move coloring inside `visibility`, if I understand it correctly it would not allow coloring per object any longer or otherwise how would the `exceptions` apply to different coloring rules? 
Coloring per object is a great feature. For example different fire classes with different colors or different sound classifications in different colors. 

Also selection is already outside `visibility`, it is inside `components`. Or maybe I misunderstood something in the message above...

Current structure:
````
"components": {
        "selection": [
            {
                "ifc_guid": "2MF28NhmDBiRVyFakgdbCT",
                "originating_system": "Example CAD Application",
                "authoring_tool_id": "EXCAD/v1.0"
            },
            {
                "ifc_guid": "0YNV9VANH3rwvGiLW5ilHo",
                "originating_system": "Example CAD Application",
                "authoring_tool_id": "EXCAD/v1.0"
            },
            {
                "ifc_guid": "0zkPUigBTCahtTAe6Cc_Na",
                "originating_system": "Example CAD Application",
                "authoring_tool_id": "EXCAD/v1.0"
            }
        ],
        "coloring": [{
            "color": "#ff0000",
            "components": [{
                "ifc_guid": "3$cshxZO9AJBebsni$z9Yk"
            }]
        }],
        "visibility": {
            "default_visbility": false,
            "exceptions": [{
                "ifc_guid": "4$cshxZO9AJBebsni$z9Yk"
            }],
            "view_setup_hints": {
                "spaces_visible": true,
                "space_boundaries_visible": false,
                "openings_visible": true
            }
        }
    }

```
May I suggest an improvement where selections are divided in selection groups where `originating_system` and `authoring_tool_id` is commonly shared. For huge selections groups these fields
are over and over repeated making the file unnecessarily large with duplicated data. Also the `ifc_guid` key can possibly be omitted for the selection, it could be an array of guid's instead.

For coloring these `originating_system` and `authoring_tool_id` fields are not present, why is this? Would it not be better to have similar interface for coloring?
I could imagine the code above would something like this:

```
"components": {
    "selection": [{
        "originating_system": "Example CAD Application",
        "authoring_tool_id": "EXCAD/v1.0",
        "components": ["2MF28NhmDBiRVyFakgdbCT", "0YNV9VANH3rwvGiLW5ilHo", "0zkPUigBTCahtTAe6Cc_Na"];
    }],
    "coloring": [{
        "color": "#ff0000",
        "originating_system": "Example CAD Application",
        "authoring_tool_id": "EXCAD/v1.0",
        "components": ["3$cshxZO9AJBebsni$z9Yk"]
    }],
    "visibility": {
        "default_visbility": false,
        "exceptions": ["4$cshxZO9AJBebsni$z9Yk"],
        "view_setup_hints": {
            "spaces_visible": true,
            "space_boundaries_visible": false,
            "openings_visible": true
        }
    }
}

```
Please don't take this as criticism to the great work you are doing for BCF, but take it as a humble suggestion for improvement. I am open for discussion. We are thinking of implementing BCF 3.0 in our application and we noticed these objects get huge for large models because of all the extra overhead.

**Comment by GeorgDangl** [comment]
Fixed with #201
